### DISCUSSION TITLE
Update errors for pre-filled forms

### DISCUSSION THREAD

**Discussion by AndGreen** [react-hook-form/react-hook-form#2846] [Ideas] [answered]
Hi, everyone!

**Where?**
I have form with _onChange mode_, _defaultValues_, and _pattern fields' validations_, but in one of the fields, defaultValues is out the pattern, so formState.isValid gonna be false, but errors object is still empty. There is a small example of such form: https://codesandbox.io/s/react-hooks-form-default-validation-no-errors-znrqb 

**What?**
The question is, can we update the errors object when validation has been failed on default values?

**Why?**
It's hard to find validation errors on huge pre-filled forms when you don't see what the problematic field because errors are empty.

Thank you for your answer!

**Comment by bluebill1049** [comment, marked as answer]
<img width="965" alt="Screen Shot 2020-09-08 at 8 51 05 am" src="https://user-images.githubusercontent.com/10513364/92419358-705e7c80-f1b0-11ea-9895-b9b524dae628.png">

isn't the error there? if you need to update error during the render of the form, you should use trigger at useEffect.
